src/v/iceberg/manifest_io.h

```diff
@@ -48,6 +50,19 @@ class manifest_io {
     upload_manifest(const manifest_path& path, const manifest&);
 
 private:
+    template<typename T>
```

Comment:
maybe this change should go first and then the previous commit could build on top of it?

Reply:
DOne